### PR TITLE
feat!: Enable license checking for flow-polymer-template

### LINKED ISSUE #15137: Add a license to polymer module in Flow

### PR DESCRIPTION
**Pull request by vursen**
## Description

The PR enables license checking for the `flow-polymer-template` module as, starting with Vaadin 24, it is only available for Prime+ customers.

Fixes #15137

## Type of change

- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/AbstractTemplate.java [hunk 1]
```diff
@@ -1,17 +1,11 @@
-/*
- * Copyright 2000-2022 Vaadin Ltd.
+/**
+ * Copyright (C) 2022 Vaadin Ltd
  *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not
- * use this file except in compliance with the License. You may obtain a copy of
- * the License at
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
```
Comment (mshabarov) [inline flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/AbstractTemplate.java:5]: Besides the headers, we have to add `LICENSE` file to the module root and update `bnd.bnd`.

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: flow-polymer-template/pom.xml [hunk 2]
```diff
@@ -13,6 +13,7 @@
     <packaging>jar</packaging>
 
     <properties>
+        <cvdlName>flow-polymer-template</cvdlName>
```
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-polymer-template/pom.xml:16]: it is used in `bnd.bnd`

[the remark this answers is not in the document]
Comment (vursen) [reply flow-polymer-template/pom.xml:16]: Done.

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java [hunk 3]
```diff
@@ -60,6 +58,19 @@
...
     static {
         UsageStatistics.markAsUsed("flow/PolymerTemplate", null);
+
+        Properties properties = new Properties();
+        try {
+            properties.load(PolymerTemplate.class
+                    .getResourceAsStream("version.properties"));
+        } catch (Exception e) {
+            LoggerFactory.getLogger(PolymerTemplate.class.getName())
+                    .warn("Unable to read the version.properties file.", e);
```
[the remark this answers is not in the document]
Comment (vursen) [reply flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java]: Good point, thanks! Done.

**Review by mshabarov — DISMISSED** [review]

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mshabarov) [reply flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java]: You lost the exception throwing.
I meant this code
```
        } catch (Exception e) {
            LoggerFactory.getLogger(PolymerTemplate.class.getName())
                    .error("Unable to read the version.properties file.");
            throw new ExceptionInInitializerError(e);
        }
```
If you won't throw, the NPE will be thrown anyway 2 lines after

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java [hunk 4]
```diff
@@ -60,6 +58,18 @@
...
     static {
         UsageStatistics.markAsUsed("flow/PolymerTemplate", null);
+
+        Properties properties = new Properties();
+        try {
+            properties.load(PolymerTemplate.class
+                    .getResourceAsStream("version.properties"));
+        } catch (Exception e) {
+            LoggerFactory.getLogger(PolymerTemplate.class.getName())
+                    .error("Unable to read the version.properties file.", e);
```
Comment (vursen) [inline flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java:68]:
```suggestion
            LoggerFactory.getLogger(PolymerTemplate.class.getName())
                    .error("Unable to read the version.properties file.");
            throw new ExceptionInInitializerError(e);
```

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (vursen) [reply flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java]: Ah, right. Fixed.

**Review by mshabarov — APPROVED** [review]

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java [hunk 5]
```diff
@@ -60,6 +58,19 @@
 
     static {
         UsageStatistics.markAsUsed("flow/PolymerTemplate", null);
+
+        Properties properties = new Properties();
+        try {
```
Comment (Artur-) [inline flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java:63]: We do not need a separate version.properties for this module. It is fine to use `com.vaadin.flow.server.Version`

**Review by Artur- — CHANGES_REQUESTED** [review]

File: flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java (hunk 5 above)
Comment (vursen) [reply flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java:63]: Good point. Let me update the PR.

Comment (vursen) [reply flow-polymer-template/src/main/java/com/vaadin/flow/component/polymertemplate/PolymerTemplate.java]: Done.

File: flow-polymer-template/pom.xml [hunk 6]
```diff
@@ -70,6 +75,15 @@
         </dependency>
     </dependencies>
     <build>
+        <resources>
```
[the remark this answers is not in the document]
Comment (vursen) [reply flow-polymer-template/pom.xml]: Done.

**Review by mshabarov — APPROVED** [review]

**Review by Artur- — APPROVED** [review]

**Pull request merged**